StripeCore/StripeCore/Source/Attestation/StripeAttest.swift

```diff
@@ -365,10 +389,43 @@ import UIKit
         }
     }
 
+    // MARK: Assertion concurrency
+
+    // Called when an assertion handle is completed
+    private func assertionCompleted() {
+        assertionInProgress = false
+
+        // Resume the next waiter if there is one
+        if !assertionWaiters.isEmpty {
+            let nextContinuation = assertionWaiters.removeFirst()
+            nextContinuation.resume()
+        }
+    }
+
     private func testmodeAssertion() async -> Assertion {
         Assertion(assertionData: Data(bytes: [0x01, 0x02, 0x03], count: 3),
                   deviceID: (try? await getDeviceID()) ?? "test-device-id",
                   appID: (try? getAppID()) ?? "com.example.test",
                   keyID: "TestKeyID")
     }
 }
+
+extension StripeAttest {
+    public class AssertionHandle {
+        public let assertion: Assertion
+        private weak var stripeAttest: StripeAttest?
+
+        init(assertion: Assertion, stripeAttest: StripeAttest) {
+            self.assertion = assertion
+            self.stripeAttest = stripeAttest
+        }
+
+        // Must be called by the caller when done with the assertion
+        public func complete() {
+            guard let stripeAttest = stripeAttest else { return }
+            Task {
+                await stripeAttest.assertionCompleted()
+            }
```

Comment:
If stripeAttest goes out of scope it seems that all the callers of stripeattest.assert() will be awaiting forever. It think is this unlikely to happen, and the correct way to implement this, but maybe we should add an stpAssert in the else clause of the guard e.g.:
```
guard let stripeAttest else {
    stpAssert(...)
    return
}
 Task {
     await ..
 } 
```

Reply:
I'll add an assert here, though I don't think it's possible for `stripeAttest` to go out of scope while it has other callers waiting on it.